lib/charms/kratos/v0/kratos_registration_web_hook.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python3
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Interface library for configuring a kratos registration webhook.
+
+The provider side is responsible for providing the configuration that kratos
+will use to call this webhook.
+
+The requirer sice (kratos) takes the configuration provided and updates its
+config.
+"""
+
+# The unique Charmhub library identifier, never change it
+import logging
+from typing import Annotated, Any, List, Optional, TypeVar, Union
+from annotated_types import T
+from ops import CharmBase, EventSource, ModelError, Object, ObjectEvents, Relation, RelationBrokenEvent, RelationCreatedEvent, RelationEvent
+from pydantic import (
+    BaseModel as _BaseModel,
+    BeforeValidator,
+    FieldSerializationInfo,
+    PlainSerializer,
+    StrictBool,
+    field_serializer,
+    field_validator,
+    Field,
+    FieldValidationInfo,
+)
+from pydantic_core import from_json
+
+
+LIBID = "37ddb4471fae41adb74299f091ee3a28"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+PYDEPS = ["pydantic"]
+
+RELATION_NAME = "kratos-registration-webhook"
+INTERFACE_NAME = "kratos_registration_webhook"
+logger = logging.getLogger(__name__)
+
+
+class BaseModel(_BaseModel):
+    def __init__(__pydantic_self__, **data: Any) -> None:
```

Comment:
question: should this init method also invoke the super class init?

```suggestion
    def __init__(__pydantic_self__, **data: Any) -> None:
        super().__init__(**data)
```

Reply:
No this is intentional, we want to change the way the validators are invoked. The parent's class init method will over-write this.